Description: mktemp temporary directory handling
 Use mktemp instead of relying that $$-$RANDOM-$RANDOM does not exist.
 firehol-1.273-CVE-2008-4953.patch gentoo patch ( https://bugs.gentoo.org/246013 )
 slightly modified by Jerome Benoit <calculs@rezozer.net>.
Origin: gentoo
Author: Robert Buchholz <rbu@goodpoint.de>
Last-Update: 2013-03-16

--- a/firehol.sh
+++ b/firehol.sh
@@ -272,8 +272,15 @@
 # ----------------------------------------------------------------------
 # Directories and files
 
-# These files will be created and deleted during our run.
-FIREHOL_DIR="/tmp/.firehol-tmp-$$-${RANDOM}-${RANDOM}"
+# Create an empty temporary directory we need for this run.
+if ! FIREHOL_DIR="`mktemp -d -t firehol-XXXXXXXXXX`"
+then
+            echo >&2
+            echo >&2
+            echo >&2 "Cannot create temporary directory."
+            echo >&2
+            exit 1
+fi
 FIREHOL_CHAINS_DIR="${FIREHOL_DIR}/chains"
 FIREHOL_OUTPUT="${FIREHOL_DIR}/firehol-out.sh"
 FIREHOL_SAVED="${FIREHOL_DIR}/firehol-save.sh"
@@ -420,20 +427,6 @@
 	"${CHMOD_CMD}" 700 "${FIREHOL_CONFIG_DIR}/services"
 fi
 
-# Remove any old directories that might be there.
-if [ -d "${FIREHOL_DIR}" ]
-then
-	"${RM_CMD}" -rf "${FIREHOL_DIR}"
-	if [ $? -ne 0 -o -e "${FIREHOL_DIR}" ]
-	then
-		echo >&2
-		echo >&2
-		echo >&2 "Cannot clean temporary directory '${FIREHOL_DIR}'."
-		echo >&2
-		exit 1
-	fi
-fi
-"${MKDIR_CMD}" "${FIREHOL_DIR}"				|| exit 1
 "${MKDIR_CMD}" "${FIREHOL_CHAINS_DIR}"			|| exit 1
 
 # prepare the file that will hold all modules to be loaded.
